packages/contracts-bedrock/test/libraries/NetworkUpgradeTxns.t.sol

```diff
@@ -0,0 +1,317 @@
+        NetworkUpgradeTxns.NetworkUpgradeTxn memory txn = NetworkUpgradeTxns.newDeploymentTx({
+            intent: _intent,
+            from: _from,
+            gas: _gas,
+            forgeArtifactPath: "GasPriceOracle.sol:GasPriceOracle"
+        });
+
+        assertEq(txn.sourceHash, NetworkUpgradeTxns.sourceHash(_intent), "sourceHash mismatch");
+        assertEq(txn.from, _from, "from mismatch");
+        assertEq(txn.to, address(0), "to should be zero for deployment");
+        assertEq(txn.mint, 0, "mint should be zero");
+        assertEq(txn.value, 0, "value should be zero");
+        assertEq(txn.gas, _gas, "gas mismatch");
+        assertFalse(txn.isSystemTransaction, "should not be system transaction");
+        assertTrue(txn.data.length > 0, "data should not be empty");
+    }
+}
+
+/// @title NetworkUpgradeTxns_WriteArtifact_Test
+/// @notice Tests the `writeArtifact` function.
+contract NetworkUpgradeTxns_WriteArtifact_Test is NetworkUpgradeTxns_TestInit {
+    /// @notice Test writeArtifact with empty array
+    function test_writeArtifact_emptyArray() public {
+        NetworkUpgradeTxns.NetworkUpgradeTxn[] memory txns = new NetworkUpgradeTxns.NetworkUpgradeTxn[](0);
+        string memory outputPath = "deployments/nut-test-empty.json";
+        NetworkUpgradeTxns.writeArtifact(txns, outputPath);
+    }
+
+    /// @notice Test writeArtifact with single Predeploy deployment
+    function test_writeArtifact_singleDeployment() public {
+        NetworkUpgradeTxns.NetworkUpgradeTxn[] memory txns = new NetworkUpgradeTxns.NetworkUpgradeTxn[](1);
+        txns[0] = NetworkUpgradeTxns.newDeploymentTx({
+            intent: INTENT_DEPLOY_L1_BLOCK,
+            from: L1_BLOCK_DEPLOYER,
+            gas: 375_000,
+            forgeArtifactPath: "L1Block.sol:L1Block"
+        });
+        string memory outputPath = "deployments/nut-test-single.json";
+        NetworkUpgradeTxns.writeArtifact(txns, outputPath);
+    }
+
+    /// @notice Test writeArtifact creates valid JSON file
+    function test_writeArtifact_succeeds() public {
+        NetworkUpgradeTxns.NetworkUpgradeTxn[] memory txns = new NetworkUpgradeTxns.NetworkUpgradeTxn[](2);
+
+        txns[0] = NetworkUpgradeTxns.newDeploymentTx({
+            intent: INTENT_DEPLOY_L1_BLOCK,
+            from: L1_BLOCK_DEPLOYER,
+            gas: 375_000,
+            forgeArtifactPath: "L1Block.sol:L1Block"
+        });
+
+        txns[1] = NetworkUpgradeTxns.newTx({
+            intent: INTENT_ENABLE_ECOTONE,
+            from: DEPOSITOR_ACCOUNT,
+            to: Predeploys.GAS_PRICE_ORACLE,
+            mint: 0,
+            value: 0,
+            gas: 50_000,
+            isSystemTransaction: false,
+            data: abi.encodeCall(IGasPriceOracle.setEcotone, ())
+        });
+
+        string memory outputPath = "deployments/nut-test.json";
+        NetworkUpgradeTxns.writeArtifact(txns, outputPath);
```

Comment:
This looks like a "smoke test", ie. as long as this line doesn't revert it succeeds, whether or not the JSON is correctly formatted.

Not necessary to address at this stage, but in a final implementation we should store the expected JSON output in a testdata dir, and then compare it to what what is created here.